datasource/amazon/ami/data.go

```diff
@@ -0,0 +1,38 @@
+//go:generate mapstructure-to-hcl2 -type DataSourceOutput
+package ami
+
+import (
+	"github.com/hashicorp/hcl/v2/hcldec"
+	"github.com/zclconf/go-cty/cty"
+)
+
+type DataSource struct {
+	config Config
+}
+
+type DataSourceOutput struct {
+	ID           string
+	Name         string
+	CreationDate string
+	Owner        string
+	OwnerName    string
+	Tags         map[string]string
+}
+
+func (d *DataSource) ConfigSpec() hcldec.ObjectSpec {
+	return d.config.FlatMapstructure().HCL2Spec()
+}
+
+func (d *DataSource) OutputSpec() hcldec.ObjectSpec {
+	return (&DataSourceOutput{}).FlatMapstructure().HCL2Spec()
+}
+
+func (d *DataSource) Configure(...interface{}) error {
+	return nil
+}
+
+func (d *DataSource) Execute() (cty.Value, error) {
+	return cty.ObjectVal(map[string]cty.Value{
+		"id": cty.StringVal("ami-0568456c"),
+	}), nil
+}
```

Comment:
Is it easy to transform a the DataSource into a cty.Value ? Will we have to add some code generation that will do that ?

Reply:
I have already thought about that, we have to do change [HCL2ValueFromConfigValue(v interface{}) cty.Value](https://github.com/hashicorp/packer/blob/master/hcl2template/shim/values.go#L92) and add the interface case there, and then move this code to the packer-plugin-sdk.

Reply:
I also thought about the code generation but not in the actual implementation 🤔  but we can make use of HCL2ValueFromConfigValue in there too.

Reply:
Oh yeah great idea, we might want to put it in the sdk then, and test it way more. 🤔 . ( and test that the output value matches the type from `OutputSpec`. )

Reply:
will add it to the todo list here!